### PR TITLE
*: refactor and add truncation metrics

### PR DESCRIPTION
**Pull request by asubiotto**

### CODE REVIEW

**Comment by asubiotto** [comment]
Friendly ping

File: metrics.go [hunk 1]
```diff
@@ -52,18 +52,22 @@ func newWALMetrics(reg prometheus.Registerer) *walMetrics {
...
+			prometheus.CounterOpts{
+				Name: "entries_truncated",
+				Help: "entries_truncated counts how many log entries have been truncated" +
+					" from the front or back.",
+			},
+			[]string{"type"},
+		),
+		truncations: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "truncations",
```
Comment (metalmatze) [inline metrics.go]:
```suggestion
				Name: "truncations_total",
```

File: metrics.go [hunk 2]
```diff
@@ -52,18 +52,22 @@ func newWALMetrics(reg prometheus.Registerer) *walMetrics {
...
-		}),
-		tailTruncations: promauto.With(reg).NewCounter(prometheus.CounterOpts{
-			Name: "tail_truncations",
-			Help: "tail_truncations counts how many log entries have been truncated" +
-				" from the tail - i.e. the newest entries. by graphing the rate of" +
-				" change over time you can see individual truncate calls as spikes.",
-		}),
+		entriesTruncated: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "entries_truncated",
```
Comment (metalmatze) [inline metrics.go]:
```suggestion
				Name: "entries_truncated_total",
```

File: metrics.go [hunk 3]
```diff
@@ -52,18 +52,22 @@ func newWALMetrics(reg prometheus.Registerer) *walMetrics {
...
+			},
+			[]string{"type"},
+		),
+		truncations: promauto.With(reg).NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "truncations",
+				Help: "truncations is the number of truncate calls categorized by whether" +
+					" the call was successful or not.",
+			},
+			[]string{"type", "success"},
```
Comment (metalmatze) [inline metrics.go:69]: The success label isn't ideal, I feel like, but I have a hard time coming up with an alternative for now. :man_shrugging:

Comment (asubiotto) [reply metrics.go:69]: Yes, we've done this elsewhere with GRPC codes for network requests, but here it's just binary whether the truncation was successful or not.

**Comment by asubiotto** [comment]
Updated.

**Review by metalmatze — APPROVED** [review]

**Pull request merged**